### PR TITLE
Parse plugin search paths from swift modules that are interactively 

### PR DESCRIPTION
**Pull request by adrian-prantl**
Parse plugin search paths from swift modules that are interactively imported in the expression evaluator or REPL.

rdar://107031048

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
Rebased

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test windows

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp [hunk 1]
```diff
@@ -1773,6 +1765,16 @@ static bool IsDWARFImported(swift::ModuleDecl &module) {
                      });
 }
 
+/// Detect whether this is a proper Swift module.
+static bool IsSerializedAST(swift::ModuleDecl &module) {
+  return std::any_of(module.getFiles().begin(), module.getFiles().end(),
+                     [](swift::FileUnit *file_unit) {
+                       return (file_unit->getKind() ==
+                               swift::FileUnitKind::SerializedAST);
+                     });
```
Comment (kastiglione) [inline lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp]: there's also `llvm::any_of` from STLExtras.h:
```suggestion
  return llvm::any_of(module.getFiles(),
                     [](swift::FileUnit *file_unit) {
                       return (file_unit->getKind() ==
                               swift::FileUnitKind::SerializedAST);
                     });
```

File: lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp [hunk 2]
```diff
@@ -2194,13 +2211,27 @@ static void ProcessModule(
...
+
+  auto ast_file_datas = module_sp->GetASTData(eLanguageTypeSwift);
+  std::string module_name = module_sp->GetSpecificationDescription();
+  LOG_PRINTF(GetLog(LLDBLog::Types), "Found %d AST file data entries in %s.",
+             (int)ast_file_datas.size(), module_name.c_str());
+  std::vector<llvm::StringRef> buffers;
+  for (auto &data : ast_file_datas)
+    if (data)
+      buffers.push_back(
+          StringRef((const char *)data->GetBytes(), data->GetByteSize()));
```
Comment (kastiglione) [inline lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp]: there are two instances of this block of code, is it worth extracting a function?

**Review by kastiglione — APPROVED** [review]

**Comment by adrian-prantl** [comment]
@swift-ci test windows

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test windows

**Comment by adrian-prantl** [comment]
@swift-ci test

**Comment by adrian-prantl** [comment]
@swift-ci test windows

**Comment by adrian-prantl** [comment]
@swift-ci test windows

**Comment by adrian-prantl** [comment]
@swift-ci test windows

**Comment by adrian-prantl** [comment]
@swift-ci test windows

**Comment by adrian-prantl** [comment]
@swift-ci test windows

**Pull request merged**